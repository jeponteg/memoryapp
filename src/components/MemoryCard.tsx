import React from "react";
import AnonymousCard from "../assets/img/anonymous-card.png";

interface MemoryCardProps {
  uuid: string;
  url: string;
  isFlipped: boolean;
  isCorrect: boolean; // Nueva prop para indicar si la carta es correcta
  onClick: () => void;
}

const MemoryCard: React.FC<MemoryCardProps> = ({ uuid, url, isFlipped, isCorrect, onClick }) => {
  return (
    <div
      className={`memory-card ${isFlipped ? "flipped" : ""} ${isCorrect ? "correct" : ""}`}
      onClick={() => !isFlipped && onClick()}
    >
      {!isFlipped ? ( // Mostrar la imagen frontal si isFlipped es true
        <div className="back border rounded-lg min-h-[330px] w-200 bg-gray-200">
          <img src={url} alt="Card" width={200} className="border rounded-lg" />
        </div>

      ) : ( // Mostrar la imagen trasera si isFlipped es false
        <div className="front relative border rounded-lg bg-gray-200 hover:bg-gray-300 cursor-pointer transition duration-300 m-5">
          <img src={AnonymousCard} alt="Card" width={200} />
        </div>
      )}
    </div>
  );
};

export default MemoryCard;
