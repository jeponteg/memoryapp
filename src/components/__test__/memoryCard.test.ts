import MemoryCard from '../MemoryCard';

describe('MemoryCard', () => {
  it('should render a flipped card correctly', () => {
    const props = {
      uuid: 'some-uuid',
      url: 'image-url.jpg',
      isFlipped: true,
      onClick: jest.fn(),
      success: false,
    };

    const component = MemoryCard(props);

    expect(component).toMatchSnapshot();
  });

  it('should render an unflipped card correctly', () => {
    const props = {
      uuid: 'some-uuid',
      url: 'image-url.jpg',
      isFlipped: false,
      onClick: jest.fn(),
      success: true,
    };

    const component = MemoryCard(props);

    expect(component).toMatchSnapshot();
  });

  it('should invoke onClick function when clicked', () => {
    const onClickMock = jest.fn();
    const props = {
      uuid: 'some-uuid',
      url: 'image-url.jpg',
      isFlipped: false,
      onClick: onClickMock,
      success: true,
    };

    const component = MemoryCard(props);

    component?.props.onClick();

    // Verificar si la función onClick se llamó correctamente
    expect(onClickMock).toHaveBeenCalled();
  });

});
