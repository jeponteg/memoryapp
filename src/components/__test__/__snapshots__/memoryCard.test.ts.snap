// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemoryCard should render a flipped card correctly 1`] = `
<div
  className="memory-card"
  onClick={[Function]}
>
  <div
    className="back border rounded-lg min-h-52 w-200 bg-gray-200 m-5 hover:bg-gray-300 cursor-pointer"
  >
    <img
      alt="Card"
      className="border rounded-lg h-52"
      src="image-url.jpg"
      width={200}
    />
  </div>
</div>
`;

exports[`MemoryCard should render an unflipped card correctly 1`] = `
<div
  className="memory-card"
  onClick={[Function]}
>
  <div
    className="back border rounded-lg min-h-52 w-200 bg-gray-200 m-5 hover:bg-gray-300 cursor-pointer"
  >
    <img
      alt="Card"
      className="border rounded-lg h-52"
      src="image-url.jpg"
      width={200}
    />
  </div>
</div>
`;
